File: visualization/python/pca_plot.py
# PCA Plot in Python for bulk RNA-Seq
import pandas as pd
import matplotlib.pyplot as plt
from sklearn.decomposition import PCA

counts = pd.read_csv('../../rna_seq/sample_counts.csv', index_col=0)
pca = PCA(n_components=2)
X_pca = pca.fit_transform(counts.T)
plt.figure(figsize=(6,5))
plt.scatter(X_pca[:,0], X_pca[:,1], s=50)
plt.xlabel(f'PC1 ({pca.explained_variance_ratio_[0]*100:.1f}% var)')
plt.ylabel(f'PC2 ({pca.explained_variance_ratio_[1]*100:.1f}% var)')
plt.title('PCA of Bulk RNA-Seq Samples')
plt.tight_layout()
plt.show()

File: visualization/r/pca_plot.R
# PCA Plot in R for bulk RNA-Seq
library(ggplot2)
library(DESeq2)
# Load DESeq2 results or count matrix
dds <- DESeqDataSetFromMatrix(countData = as.matrix(read.csv("../../rna_seq/sample_counts.csv", row.names=1)),
                              colData = DataFrame(condition=rep("A", ncol(read.csv("../../rna_seq/sample_counts.csv"))-1)),
                              design = ~1)
dds <- estimateSizeFactors(dds)
dds <- estimateDispersions(dds)
dds <- DESeq(dds)
vsd <- vst(dds)
pcaData <- plotPCA(vsd, intgroup = c("condition"), returnData=TRUE)
percentVar <- round(100 * attr(pcaData, "percentVar"))
ggplot(pcaData, aes(PC1, PC2)) +
  geom_point(size=3) +
  xlab(paste0("PC1: ", percentVar[1], "% variance")) +
  ylab(paste0("PC2: ", percentVar[2], "% variance")) +
  theme_minimal()
